pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercurySampleDao.java

```diff
@@ -0,0 +1,156 @@
+package org.broadinstitute.dsm.db.dao.mercury;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.dao.Dao;
+import org.broadinstitute.dsm.db.dto.mercury.MercurySampleDto;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.lddp.db.SimpleResult;
+
+public class MercurySampleDao implements Dao<MercurySampleDto> {
+    public static String SQL_GET_ELIGIBLE_TISSUES =
+            "SELECT * FROM ("
+                    + "SELECT collaborator_sample_id, t.sent_gp, oD.date_px, oD.tissue_received, t.tissue_id    "
+                    + "FROM  ddp_participant as p    "
+                    + "LEFT JOIN ddp_instance as ddp on (ddp.ddp_instance_id = p.ddp_instance_id)    "
+                    + "LEFT JOIN ddp_institution inst on  (inst.participant_id = p.participant_id)    "
+                    + "LEFT JOIN ddp_medical_record mr on (mr.institution_id = inst.institution_id AND NOT mr.deleted <=> 1)    "
+                    + "LEFT JOIN ddp_onc_history_detail oD on (mr.medical_record_id = oD.medical_record_id AND NOT oD.deleted <=> 1)    "
+                    + "LEFT JOIN ddp_tissue t on (oD.onc_history_detail_id = t.onc_history_detail_id AND NOT t.deleted <=> 1)    "
+                    + "WHERE oD.tissue_received IS NOT NULL AND p.ddp_participant_id = ? AND ddp.instance_name = ? "
+                    + "AND  IFNULL(t.uss_count, 0) = (SELECT count(*) from sm_id sm "
+                    + "left join sm_id_type smtype on (sm.sm_id_type_id = smtype.sm_id_type_id) "
+                    + "where smtype.sm_id_type = \"uss\" and sm.tissue_id = t.tissue_id) "
+                    + "AND "
+                    + "IFNULL(t.scrolls_count, 0) = (SELECT count(*) from sm_id sm "
+                    + "left join sm_id_type smtype on (sm.sm_id_type_id = smtype.sm_id_type_id) "
+                    + "where smtype.sm_id_type = \"scrolls\" and sm.tissue_id = t.tissue_id ) ) as table1 "
+                    + "left join (select max(order_date) as order_date, tissue_id as seqt "
+                    + "From ddp_mercury_sequencing where ddp_participant_id = ? group by (tissue_id)   ) as table2 "
+                    + "on table2.seqt = table1.tissue_id ";
+
```

Comment:
@SimoneMaiwald see here

Reply:
in easy words: when is a tissue eligible?

Reply:
Tissue should have collection date as date_px, be received and sent to GP,  and have all sm ids received, also number of uss and scrolls numbers should match the number of those inserted ( like if they entered 20 uss, they should have entered all the 20 uss values)

Reply:
tissue is the complex one, I am doing all of that in sql now